clang/lib/Sema/BoundsWideningAnalysis.cpp

```diff
@@ -785,8 +806,9 @@ void BoundsWideningAnalysis::GetVarsAndBoundsInPtrDeref(
 
     RangeBoundsExpr *WidenedBounds =
       new (Ctx) RangeBoundsExpr(R->getLowerExpr(),
-                                BWUtil.AddOffsetToExpr(DerefExpr, 1),
-                                SourceLocation(), SourceLocation());
+        BWUtil.AddOffsetToExpr(EB->TermCondInfo.DerefExpr, 1),
+        SourceLocation(), SourceLocation());
```

Comment:
I am not sure if it would be more meaningful to use the `SourceLocation` of the widen expr here.
I face the similar question.